.dev/docker/env/Dockerfile

```diff
@@ -36,44 +34,48 @@ RUN apt-get update \
       libusb-1.0-0-dev \
       libvtk${VTK_VERSION}-dev \
       libvtk${VTK_VERSION}-qt-dev \
+      lsb-release \
       qtbase5-dev \
       software-properties-common \
+      wget \
+      xvfb \
+  && if [ "$(lsb_release -sr)" = "18.04" ]; then \
+       apt-get -V install -y nvidia-cuda-toolkit g++-6 ; \
+     else \
+       apt-get -V install -y nvidia-cuda-toolkit-gcc ; \
+     fi \
  && rm -rf /var/lib/apt/lists/*
 
-# Eigen patch (https://eigen.tuxfamily.org/bz/show_bug.cgi?id=1462) to fix issue metioned
-# in https://github.com/PointCloudLibrary/pcl/issues/3729 is available in Eigen 3.3.7
-# Not needed from 20.04 since it is the default version from apt
-RUN if [ `pkg-config --modversion eigen3 | cut -d. -f3` -lt 7 ]; then \
-     wget -qO- https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.gz | tar xz \
-     && apt install -y libblas-dev \
-     && cd eigen-3.3.7 \
+# Environment variable selected by CMake 3.13 and above to select the CUDA compiler.
+ENV CUDAHOSTCXX=/usr/lib/nvidia-cuda-toolkit/bin/g++
+
+# Use libeigen3-dev if it meets the minimal version.
+# In most cases libeigen3-dev is already installed as a dependency of libvtk6-dev & libvtk7-dev, but not in every case (libvtk9 doesn't seem to have this dependency),
+# so install it from apt if the version is sufficient.
+RUN if dpkg --compare-versions $(apt-cache show --no-all-versions libeigen3-dev | grep -P -o 'Version:\s*\K.*') ge ${EIGEN_MINIMUM_VERSION}; then \
+     apt-get -V install -y libeigen3-dev ; \
+    else \
+     wget -qO- https://gitlab.com/libeigen/eigen/-/archive/${EIGEN_MINIMUM_VERSION}/eigen-${EIGEN_MINIMUM_VERSION}.tar.gz | tar xz \
+     && cd eigen-${EIGEN_MINIMUM_VERSION} \
      && mkdir build \
      && cd build \
      && cmake .. \
      && make install \
      && cd ../.. \
-     && rm -rf eigen-3.3.7/ \
-     && rm -f eigen-3.3.7.tar.gz ; \
-    fi
-
-# To avoid CUDA build errors on CUDA 9.2+ GCC 7 is required
-RUN if [ `gcc -dumpversion | cut -d. -f1` -lt 7 ]; then add-apt-repository ppa:ubuntu-toolchain-r/test \
-     && apt update \
-     && apt install g++-7 -y \
-     && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 60 --slave /usr/bin/g++ g++ /usr/bin/g++-7 \
-     && update-alternatives --config gcc ; \
+     && rm -rf eigen-${EIGEN_MINIMUM_VERSION}/ \
+     && rm -f eigen-${EIGEN_MINIMUM_VERSION}.tar.gz ; \
```

Comment:
Do we have this file? L58 un-archives the tar directly.

Reply:
You are right, I added a call to `ls -la` before the `rm` and got:
```
drwxr-xr-x   1 root root 4096 May 19 20:36 .
drwxr-xr-x   1 root root 4096 May 19 20:36 ..
-rwxr-xr-x   1 root root    0 May 19 20:36 .dockerenv
drwxr-xr-x   1 root root 4096 May 18 13:52 bin
drwxr-xr-x   2 root root 4096 Apr 24  2018 boot
drwxr-xr-x   5 root root  340 May 19 20:36 dev
drwxrwxr-x  15 root root 4096 May 19 20:36 eigen-3.3.7
drwxr-xr-x   1 root root 4096 May 19 20:36 etc
drwxr-xr-x   2 root root 4096 Apr 24  2018 home
drwxr-xr-x   1 root root 4096 May 18 13:53 lib
drwxr-xr-x   2 root root 4096 May 18 13:52 lib32
drwxr-xr-x   2 root root 4096 Apr 16 14:53 lib64
drwxr-xr-x   2 root root 4096 Apr 16 14:45 media
drwxr-xr-x   2 root root 4096 Apr 16 14:45 mnt
drwxr-xr-x   2 root root 4096 Apr 16 14:45 opt
dr-xr-xr-x 260 root root    0 May 19 20:36 proc
drwx------   1 root root 4096 May 19 20:36 root
drwxr-xr-x   1 root root 4096 May 18 13:51 run
drwxr-xr-x   1 root root 4096 May 18 13:53 sbin
drwxr-xr-x   2 root root 4096 Apr 16 14:45 srv
dr-xr-xr-x  13 root root    0 May 19 20:36 sys
drwxrwxrwt   1 root root 4096 May 19 20:36 tmp
drwxr-xr-x   1 root root 4096 May 18 13:52 usr
drwxr-xr-x   1 root root 4096 Apr 16 14:54 var
```